Handlers/ApacheHandlerSimple.lua

```diff
@@ -0,0 +1,92 @@
+--      SetEnv  QUEUEIT_SECRET_KEY      "{SECRET_KEY}"
+--      SetEnv  QUEUEIT_INT_CONF_FILE   "{APP_FOLDER}/integration_config.json"
+--      LuaMapHandler  "{URI_PATTERN}"  "{APP_FOLDER}/Handlers/ApacheHandlerSimple.lua"
+--      LuaPackagePath "{APP_FOLDER}/SDK/?.lua"
+--      LuaPackagePath "{APP_FOLDER}/Helpers/?/?.lua"
+--      LuaPackagePath "{APP_FOLDER}/Handlers/?.lua"
+--
+-- AUTHOR: Simon Studer (mail@studer.si)
+--
+-- LICENSE: Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
+--
+-- VERSION: 1 (2020-01-10)
+----------------------------------------------------------------------------------------------------
+
+
+local DEBUG_TAG = "ApacheHandlerHelperSimple.lua"
+
+local kuHandler = require("KnownUserApacheHandler")
+local file = require("file")
+
+local function initRequiredHelpers(r)
+    local iHelpers = require("KnownUserImplementationHelpers")
+
+    iHelpers.request.getAbsoluteUri = function()
+        local fullUrl = string.format("%s://%s:%s%s",
+            r.is_https and "https" or "http",
+            r.hostname,
+            r.port,
+            r.unparsed_uri)
+        r:debug(string.format("[%s] Rebuilt request URL as: %s", DEBUG_TAG, fullUrl))
+        return fullUrl
+    end
+end
+
+function handle(r)
+
+    -- catch errors if any occur
+    local success, result = pcall(function()
+
+        -- get configuration from environment variables
+        local customerId = r.subprocess_env["QUEUEIT_CUSTOMER_ID"]
+        local secretKey = r.subprocess_env["QUEUEIT_SECRET_KEY"]
+        local intConfFile = r.subprocess_env["QUEUEIT_INT_CONF_FILE"]
+        r:debug(string.format("[%s] Environment variable QUEUEIT_CUSTOMER_ID: %s", DEBUG_TAG, customerId))
+        r:debug(string.format("[%s] Environment variable QUEUEIT_SECRET_KEY: %s", DEBUG_TAG, secretKey))
+        r:debug(string.format("[%s] Environment variable QUEUEIT_INT_CONF_FILE: %s", DEBUG_TAG, intConfFile))
+        assert(customerId ~= nil, "customerId invalid")
+        assert(secretKey ~= nil, "secretKey invalid")
+        assert(intConfFile ~= nil, "config invalid")
+
+        -- initialize helper functions
+        initRequiredHelpers(r)
+
+        -- read integration configuration from file
+        local intConfJson = file.readAll(intConfFile)
+        r:debug(string.format("[%s] Content of file %s: %s", DEBUG_TAG, intConfFile, intConfJson))
+
+        -- return the result from handling the request by the SDK
+        return kuHandler.handleByIntegrationConfig(customerId, secretKey, intConfJson, r)
+    end)
+
+    -- check if error occurred during request handling
+    if (success) then
+        r:debug(string.format("[%s] Request handling successful: result => %s", DEBUG_TAG, result))
+        return result
+    else
+        r:err(string.format("[%s] Request handling not successful (denying access): error => %s", DEBUG_TAG, result))
+        return 400 -- Bad Request (something must be wrong with the request)
```

Comment:
In our sample code we return apache2.DECLINED in case of errors, so that request process handling can continue.
The argument goes "It's better to disable the queue if there is an error than disabling the entire site."

Reply:
My thought was that if you find a way to trigger an error, then you can easily bypass the queue. So the application would be safe by default and only if you prove you have been through the queue are you allowed to access the application.

But I definitely see the appeal of making sure the application stays available. I have added a commit which adds a configuration item that allows the user to easily change the behaviour via the Apache configuration.

The default behaviour is to decline as before but by setting the env variable `QUEUEIT_ERROR_CODE`, an HTTP code can be used instead.

Reply:
Why I recommend the decline option is not because of a user sending a request that makes the code break, but that the provided config file is invalid. 
If this is the case, then 400 BadRequest response is not optimal. But again it depends on your setup what is most important: "queue-it" or "site" should always work.

Reply:
Yes, it really depends where the priorities lie. I think the changes I added take that into effect nicely. By default, apache2.DECLINED is returned. Then, the behaviour can be changed by setting an error code using the QUEUEIT_ERROR_CODE environment variable in `httpd.conf`.